cagov/ui-claim-tracker
# Adds Shimmer Loading Effect

### Description

  - Adds a shimmer component & styling for it
  - Adds shimmerable components for:
    - Key&value fields
    - Text lines
    - Next Steps block
  - Embeds those components in the Claim Section where appropriate
  - Passes loading state down through the component stack from where we make the API call
    - **nb:** its possible there's a more elegant way to do this, but I'm calling Good Enough for now with our looming deadlines
  - Adds stories for each

### Changes
Component | Normal | Loading
-- | -- | --
Main | ![main app normal](https://user-images.githubusercontent.com/6129479/121380910-47237600-c913-11eb-8eef-8a0c3e695424.png) | ![main app loading](https://user-images.githubusercontent.com/6129479/121380766-278c4d80-c913-11eb-9dfc-279de6817501.png)
Claim Details | ![image](https://user-images.githubusercontent.com/6129479/121381183-7cc85f00-c913-11eb-88a8-724b4ae4885f.png) | ![image](https://user-images.githubusercontent.com/6129479/121381232-894cb780-c913-11eb-8882-2b6b94a630be.png)
Claim Status | ![image](https://user-images.githubusercontent.com/6129479/121381276-94074c80-c913-11eb-92ab-8e38d8f59e20.png) | ![image](https://user-images.githubusercontent.com/6129479/121381316-9d90b480-c913-11eb-82cb-667d179008d7.png)
NextSteps | ![image](https://user-images.githubusercontent.com/6129479/121381398-b0a38480-c913-11eb-9e90-727f2d732e82.png) | ![image](https://user-images.githubusercontent.com/6129479/121381435-b8fbbf80-c913-11eb-9759-58d5f033f94f.png)
InfoField | ![image](https://user-images.githubusercontent.com/6129479/121381485-c3b65480-c913-11eb-8d44-03dfb7b9d7aa.png) | ![image](https://user-images.githubusercontent.com/6129479/121381537-cdd85300-c913-11eb-8ca8-65337813c721.png)
TextLine | ![image](https://user-images.githubusercontent.com/6129479/121381584-d761bb00-c913-11eb-98d3-b60dcaf92e23.png) | ![image](https://user-images.githubusercontent.com/6129479/121381619-df215f80-c913-11eb-924c-460016f9e2a4.png)
Shimmer | - | ![image](https://user-images.githubusercontent.com/6129479/121381646-e6486d80-c913-11eb-9a55-6796894083e8.png)
ShimmerField | - | ![image](https://user-images.githubusercontent.com/6129479/121381680-ed6f7b80-c913-11eb-8274-f8dc328f7067.png)

### Design Review
The Storybook for Design's review can be accessed [here.](https://60705d04dcad7600211e34d2-kfvehegept.chromatic.com)
See the Page Section Main component, and its Loading state.

### Testing
(tests should pass after #219 merges)
The loading state cannot be invoked naturally the app until we split the page load into two. For now

  - `git fetch && git checkout kat/121_shimmer_loading && git pull`
  - `yarn dev`
  - Navigate to http://localhost:3000/claimstatus - App should load without issue.
  - edit `pages/index.tsx` line 37 to  `loading={true}`
  - Navigate to http://localhost:3000/claimstatus - App should load with the shimmers.

You can also examine the various componenets & page sections listed above in Storybook and toggle Loading to see each one.
  - `git fetch && git checkout kat/121_shimmer_loading && git pull`
  - `yarn storybook`
  
Resolves #121 

- [ ] Claim Status shows a shimmer for pending data
- [ ] Ticket filed for splitting page load
